meta/sai_meta.cpp

```diff
@@ -5926,7 +5926,110 @@ sai_status_t meta_sai_get_stats_oid(
 
     status = get_stats(object_type, object_id, stats_enum, count, counter_id_list, counter_list);
 
-    META_LOG_STATUS(status);
+    META_LOG_STATUS(get, status);
+
+    return status;
+}
+
+sai_status_t meta_generic_validation_clear_stats(
+        _In_ const sai_object_meta_key_t& meta_key,
```

Comment:
meta_key should be checkd whether it's oid object ID, and whether this object Id is present in local db, to prevent user putting here arbitrary object, and also there should be additional check whether this object type is matching stats type, for example if port object is not getting router interface stats etc

Reply:
Adding validation infrastructure is not the focus of the PR. The focus here is to establish the op path from orchagent to the ASIC, leveraging the validation that is already there.

1) Check whether it's oid object ID
This is done in a call to meta_sai_validate_oid() https://github.com/Azure/sonic-sairedis/pull/463/files#diff-3e86685996f853406e8d0f177ef07fbdR6004

2) Whether this object Id is present in local db, to prevent user putting here arbitrary object
This should be handled in syncd that has visibility to local db, and specifically it is done at the return of a call to translate_vid_to_rid(). See new commits to address the validation https://github.com/Azure/sonic-sairedis/pull/463/commits/754a3b819e75c5f2526c9e1d0f26dc5ee665d5e0

3) whether this object type is matching stats type, for example if port object is not getting router interface stats etc
This is what this function does---checking the counter id list against sai_metadata_enum, which is chosen based on the object type passed in
https://github.com/Azure/sonic-sairedis/pull/463/files#diff-eab6aa45f3b099e81c2c3026184ae12fR194